src/amp-story-player/amp-story-player-impl.js

```diff
@@ -705,20 +731,45 @@ export class AmpStoryPlayer {
    * @param {number} storyDelta
    */
   go(storyDelta) {
+    if (storyDelta === 0) {
+      return;
+    }
     if (
-      this.currentIdx_ + storyDelta >= this.stories_.length ||
-      this.currentIdx_ + storyDelta < 0
+      this.element_.getAttribute('enable-circular-wrapping') !== 'true' &&
+      this.isIndexOutofBounds_(storyDelta)
     ) {
       throw new Error('Out of Story range.');
     }
-    if (storyDelta === 0) {
+    if (!this.isIndexOutofBounds_(storyDelta)) {
+      const currentStory = this.stories_[this.currentIdx_ + storyDelta];
+      this.show(currentStory.href);
+      this.signalNavigation_();
+      return;
+    }
+    if (
+      this.element_.getAttribute('enable-circular-wrapping') === 'true' &&
+      this.isIndexOutofBounds_(storyDelta) &&
+      storyDelta > 0
+    ) {
+      const currentStory = this.stories_[
```

Comment:
we can extract this logic into a variable depending on the delta and then call the shared methods at the end:

```
const currentStory = 
  storyDelta > 0 ? 
      this.stories_[this.currentIdx_ + storyDelta - this.stories_.length] : 
      this.stories_[this.currentIdx_ + storyDelta + this.stories_.length];


this.show(currentStory.href);
this.signalNavigation_();
```

or if you want to go fancy:
```
const currentStory = this.stories_[this.currentIdx_ + storyDelta % this.stories_.length];

this.show(currentStory.href);
this.signalNavigation_();
```